### PR TITLE
Fix/2204 prisma broken query

### LINKED ISSUE #2230: 🐛 Bug Report: Generated App - query for /api/entities/{id}/sub-entities is missing pagination
### What happened?

for example, we have two entities - posts, and tags. Where each post can have multiple tags.
We have two different endpoints to get a list of tags

/api/tags
/api/posts/{id}/tags

the first endpoint is using `TagFindManyArgs` to filter the tags which includes `where`, `orderBy`, `Skip`, and `Take`
the first endpoint is using `TagWhereInput` to filter the tags which only includes the inner properties for `where`

They both should use `TagFindManyArgs`

### What you expected to happen

The two endpoints should use the same set of query parameters

### How to reproduce

create an app with two entities with relation and open the generated REST API code

### Amplication version

0.11.2

### Environment

n/a

### Anything else

_No response_

### Are you willing to submit PR?

_No response_

### Code of Conduct

I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by abrl91**
#2204 
This pr would also fix #2230 
The issue on nestjs/swagger repo: https://github.com/nestjs/swagger/issues/1803

A bug in nestjs/swagger generated the swagger UI in an incorrect way that cause the API request to fail. 

[[ @abrl91 please add a link to the issue you opened on the nestjs repo ]]

To fix this issue a new decorator (@ApiNestedQuery) was created to generate the nested properties of the query type as expected. 

In case this issue will be resolved in the future by the Nestjs team, we can remove the local implementation of ApiNestedQuery

### CODE REVIEW

File: packages/amplication-data-service-generator/src/server/resource/controller/to-many.template.ts [hunk 1]
```diff
@@ -7,16 +7,21 @@ import * as defaultAuthGuard from "../auth/defaultAuth.guard";
...
+// @ts-ignore
+import {ApiNestedQuery} from "../../decorators/api-nested-query.decorator";
+import {plainToClass} from "class-transformer";
 
 declare interface WHERE_UNIQUE_INPUT {
   id: string;
 }
 declare interface RELATED_ENTITY_WHERE_UNIQUE_INPUT {}
-declare interface RELATED_ENTITY_WHERE_INPUT {}
+declare class RELATED_ENTITY_WHERE_INPUT {}
```
Comment (yuval-hazaz) [inline packages/amplication-data-service-generator/src/server/resource/controller/to-many.template.ts:18]: This class is not being used and can be removed

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]
Great job Amit!!!

please see the comment for a minor change

**Comment by yuval-hazaz** [comment]
@abrl91 please add a link to the issue you opened on the nestjs repo in the description of the PR above

File: packages/amplication-data-service-generator/src/server/resource/controller/to-many.template.ts (hunk 1 above)
Comment (abrl91) [reply packages/amplication-data-service-generator/src/server/resource/controller/to-many.template.ts:18]: This is weird. It's class on my file
image

@yuval-hazaz

Comment (abrl91) [reply packages/amplication-data-service-generator/src/server/resource/controller/to-many.template.ts:18]: What should I do with line 28?
image

Comment (yuval-hazaz) [reply packages/amplication-data-service-generator/src/server/resource/controller/to-many.template.ts:18]: you are correct, my mistake 👍

**Review by yuval-hazaz — APPROVED** [review]

**Pull request merged**